packages/public-api/src/website/service.ts

```diff
@@ -40,38 +40,41 @@ export async function findOrCreateWebPage(url: URL, em: EntityManager) {
   return webPageEntity;
 }
 
-export async function getOrRequestWebPageScan(url: string) {
+export async function getWebPageScan(url: string) {
   const parsedUrl = new URL(url);
-
   const db = await getDatabaseConnection();
 
-  const result = await db.transaction(async (em) => {
-    const webPageScanRepo = em.getRepository(WebPageScan);
+  const webPageScanRepo = db.getRepository(WebPageScan);
+  const webPageEntity = await findOrCreateWebPage(parsedUrl, db.createEntityManager());
 
-    const webPageEntity = await findOrCreateWebPage(parsedUrl, em);
+  return await webPageScanRepo
+    .createQueryBuilder('scan')
+    .where('scan.web_page_id = :webPageId', { webPageId: webPageEntity.id })
+    .orderBy('scan.created_at', 'DESC')
+    .limit(1)
+    .getOne();
+}
 
-    const mostRecentScan = await webPageScanRepo
-      .createQueryBuilder('scan')
-      .where('scan.web_page_id = :webPageId', { webPageId: webPageEntity.id })
-      .orderBy('scan.created_at', 'DESC')
-      .limit(1)
-      .getOne();
+export async function requestWebPageRescan(url: string) {
+  const parsedUrl = new URL(url);
+  const db = await getDatabaseConnection();
+  const lastScan = await getWebPageScan(url);
+  if (lastScan && Date.now() - lastScan?.createdAt.getTime() < RESCAN_TIMEOUT_MS) {
+    return false;
```

Comment:
I'd still resort to simply returning an older scan if timeout hasn't yet passed. And had two fns like `getWebPageScan` and `getOrRequestWebPageScan`.

Reply:
We don't use any data returned from `requestWebPageRescan` and we don't call `getWebPageScan` subsequently, so I don't see any reason why it should return any data.

Reply:
I'd suggest to use only `getWebPageScan` for now and make a separate `rescan` button that will trigger `requestWebPageRescan`

Reply:
We can merge this as-is, https://github.com/gradejs/gradejs/pull/65 covers the issue